author: Andres Salomon <dilinger@debian.org>
description: work around a clang-13 issue...

../../components/password_manager/core/browser/affiliation/affiliation_fetcher_base.cc:182:6: error: equality comparison operator can only be defaulted in a class definition
bool operator==(const AffiliationFetcherInterface::RequestInfo& lhs,
     ^
1 error generated.


Just move the default assignment from the class definitions
to the class declarations. This isn't necessary with clang-14.

--- a/components/password_manager/core/browser/affiliation/affiliation_fetcher_base.cc
+++ b/components/password_manager/core/browser/affiliation/affiliation_fetcher_base.cc
@@ -185,7 +185,5 @@ void AffiliationFetcherBase::OnSimpleLoa
   }
 }
 
-bool operator==(const AffiliationFetcherInterface::RequestInfo& lhs,
-                const AffiliationFetcherInterface::RequestInfo& rhs) = default;
 
 }  // namespace password_manager
--- a/components/password_manager/core/browser/affiliation/affiliation_fetcher_base.h
+++ b/components/password_manager/core/browser/affiliation/affiliation_fetcher_base.h
@@ -80,9 +80,6 @@ class AffiliationFetcherBase : public vi
   std::unique_ptr<network::SimpleURLLoader> simple_url_loader_;
 };
 
-bool operator==(const AffiliationFetcherInterface::RequestInfo& lhs,
-                const AffiliationFetcherInterface::RequestInfo& rhs);
-
 }  // namespace password_manager
 
 #endif  // COMPONENTS_PASSWORD_MANAGER_CORE_BROWSER_AFFILIATION_AFFILIATION_FETCHER_BASE_H_
--- a/components/password_manager/core/browser/affiliation/affiliation_fetcher_interface.h
+++ b/components/password_manager/core/browser/affiliation/affiliation_fetcher_interface.h
@@ -20,7 +20,7 @@ class AffiliationFetcherInterface {
     bool change_password_info = false;
     bool psl_extension_list = false;
 
-    friend bool operator==(const RequestInfo&, const RequestInfo&);
+    friend bool operator==(const RequestInfo&, const RequestInfo&) = default;
   };
 
   AffiliationFetcherInterface() = default;
--- a/components/password_manager/core/browser/password_form.cc
+++ b/components/password_manager/core/browser/password_form.cc
@@ -242,7 +242,6 @@ AlternativeElement& AlternativeElement::
 
 AlternativeElement::~AlternativeElement() = default;
 
-bool AlternativeElement::operator==(const AlternativeElement&) const = default;
 
 bool AlternativeElement::operator<(const AlternativeElement& other) const {
   return std::tie(value, field_renderer_id, name) <
--- a/components/password_manager/core/browser/password_form.h
+++ b/components/password_manager/core/browser/password_form.h
@@ -45,7 +45,7 @@ struct AlternativeElement {
   AlternativeElement& operator=(AlternativeElement&& rhs);
   ~AlternativeElement();
 
-  bool operator==(const AlternativeElement&) const;
+  bool operator==(const AlternativeElement&) const = default;
   bool operator<(const AlternativeElement&) const;
 
   // The value of the field.
--- a/components/password_manager/core/browser/passkey_credential.cc
+++ b/components/password_manager/core/browser/passkey_credential.cc
@@ -99,7 +99,5 @@ std::u16string PasskeyCredential::GetAut
   return l10n_util::GetStringUTF16(id);
 }
 
-bool operator==(const PasskeyCredential& lhs,
-                const PasskeyCredential& rhs) = default;
 
 }  // namespace password_manager
--- a/components/password_manager/core/browser/passkey_credential.h
+++ b/components/password_manager/core/browser/passkey_credential.h
@@ -107,7 +107,6 @@ class PasskeyCredential {
   absl::optional<std::u16string> authenticator_label_;
 };
 
-bool operator==(const PasskeyCredential& lhs, const PasskeyCredential& rhs);
 
 }  // namespace password_manager
 
--- a/net/cert/x509_cert_types.cc
+++ b/net/cert/x509_cert_types.cc
@@ -17,7 +17,6 @@ CertPrincipal::CertPrincipal(CertPrincip
 
 CertPrincipal::~CertPrincipal() = default;
 
-bool CertPrincipal::operator==(const CertPrincipal& other) const = default;
 
 bool CertPrincipal::EqualsForTesting(const CertPrincipal& other) const {
   return *this == other;
--- a/net/cert/x509_cert_types.h
+++ b/net/cert/x509_cert_types.h
@@ -54,7 +54,7 @@ struct NET_EXPORT CertPrincipal {
  private:
   // Comparison operator is private and only defined for use by
   // EqualsForTesting, see comment there for more details.
-  bool operator==(const CertPrincipal& other) const;
+  bool operator==(const CertPrincipal& other) const = default;
 };
 
 }  // namespace net
--- a/chrome/browser/web_applications/web_app.cc
+++ b/chrome/browser/web_applications/web_app.cc
@@ -886,10 +886,6 @@ void WebApp::IsolationData::SetPendingUp
   pending_update_info_ = pending_update_info;
 }
 
-bool WebApp::IsolationData::PendingUpdateInfo::operator==(
-    const WebApp::IsolationData::PendingUpdateInfo& other) const = default;
-bool WebApp::IsolationData::PendingUpdateInfo::operator!=(
-    const WebApp::IsolationData::PendingUpdateInfo& other) const = default;
 
 bool WebApp::operator==(const WebApp& other) const {
   auto AsTuple = [](const WebApp& app) {
--- a/chrome/browser/web_applications/web_app.h
+++ b/chrome/browser/web_applications/web_app.h
@@ -347,8 +347,8 @@ class WebApp {
       PendingUpdateInfo(const PendingUpdateInfo&);
       PendingUpdateInfo& operator=(const PendingUpdateInfo&);
 
-      bool operator==(const PendingUpdateInfo&) const;
-      bool operator!=(const PendingUpdateInfo&) const;
+      bool operator==(const PendingUpdateInfo&) const = default;
+      bool operator!=(const PendingUpdateInfo&) const = default;
 
       base::Value AsDebugValue() const;
       friend std::ostream& operator<<(std::ostream& os,
--- a/components/cloud_devices/common/printer_description.cc
+++ b/components/cloud_devices/common/printer_description.cc
@@ -1073,7 +1073,6 @@ MediaType::MediaType(const std::string&
                      const std::string& custom_display_name)
     : vendor_id(vendor_id), custom_display_name(custom_display_name) {}
 
-bool MediaType::operator==(const MediaType& other) const = default;
 
 bool MediaType::IsValid() const {
   return !vendor_id.empty();
--- a/components/cloud_devices/common/printer_description.h
+++ b/components/cloud_devices/common/printer_description.h
@@ -574,7 +574,7 @@ struct MediaType {
   MediaType(const std::string& vendor_id,
             const std::string& custom_display_name);
 
-  bool operator==(const MediaType& other) const;
+  bool operator==(const MediaType& other) const = default;
   bool operator!=(const MediaType& other) const { return !(*this == other); }
 
   bool IsValid() const;
